src/pages/en/core-concepts/astro-components.md

```diff
@@ -143,6 +143,35 @@ const name = "Astro";
 <MyComponent templateLiteralNameAttribute={`MyNameIs${name}`} />
 ```
 
+:::caution
+HTML attributes will be converted to strings, so functions and objects don't work when passed to HTML elements.
+For example, you can't assign an event handler to an HTML element in an Astro component:
+
+```astro
+---
+function handleClick () {
+    console.log("button clicked!");
+}
+// This doesn't work!
+---
+<button onClick={handleClick}>Nothing will happen when you click me!</button>
+```
```

Comment:
A bit nit-picky, but wondering if we can do more to highlight that this is the BROKEN example:

````suggestion
```astro
---
// dont-do-this.astro
function handleClick () {
    console.log("button clicked!");
}
---
<!-- ❌ This doesn't work! ❌ -->
<button onClick={handleClick}>Nothing will happen when you click me!</button>
```
````